girder-dandi-archive/girder_dandi_archive/util.py

```diff
@@ -23,3 +28,46 @@ def validate_dandiset_identifier(dandiset_identifier):
 
 def get_or_create_drafts_collection():
     return Collection().createCollection(DANDI_DRAFTS_COLLECTION_NAME, reuseExisting=True)
+
+
+def find_dandiset_by_identifier(identifier):
+    """Use a unique identifier to find a dandiset in the dandiset drafts collection."""
+    drafts = get_or_create_drafts_collection()
+    return Folder().findOne({"parentId": drafts["_id"], "meta.dandiset.identifier": identifier})
+
+
+def dandiset_find(query, **kwargs):
```

Comment:
can you remove this? it isn't called anywhere and it isn't clear to me how to call it or why it is needed from reading the code (I'm not saying it is not useful, but it is opaque to me when to call it and how, and that's from reading it in context of the PR that adds it. in the future someone coming in will not know how to use it, but will be afraid to remove it). i think it just adds cruft to the codebase and violates YAGNI.